### PR TITLE
feat(argon2): add `secret` parameter

### LINKED ISSUE #53: Feature request: Add `secret` parameter to argon2 implementation
<https://github.com/P-H-C/phc-winner-argon2#library> mentions additional parameters, among them:

> The secret parameter, which is used for [keyed hashing](https://en.wikipedia.org/wiki/Hash-based_message_authentication_code). This allows a secret key to be input at hashing time (from some external location) and be folded into the value of the hash. This means that even if your salts and hashes are compromised, an attacker cannot brute-force to find the password without the key.

### PR DESCRIPTION
**Pull request by malobre**
Closes #53.

### CODE REVIEW

**Comment by malobre** [comment]
@Daninet For the tests, should I add specific tests for the secret parameter or just integrate it in the existing ones ?

File: lib/argon2.ts [hunk 1]
```diff
@@ -37,6 +37,10 @@ export interface IArgon2Options {
    * Desired output type. Defaults to 'hex'
    */
   outputType?: 'hex' | 'binary' | 'encoded';
+  /**
+   * Secret, an optional key that will be folded into the value of the hash.
+   */
+  secret?: IDataType;
```
Comment (malobre) [inline lib/argon2.ts:43]: If you think of a better / easier to understand explanation I'm all ears

Comment (malobre) [reply lib/argon2.ts:43]: Maybe "secret for keyed hashing" ?

**Comment by Daninet** [comment]
Awesome. You can create a new test group dedicated for the secret parameter with a few test cases here: https://github.com/Daninet/hash-wasm/blob/master/test/argon2.test.ts 

Thank you!

**Comment by malobre** [comment]
- [x] `npm run eslint`
- [x] `npm run test`

**Comment by malobre** [comment]
@Daninet friendly bump :)

**Review by Daninet — APPROVED** [review]
Thank you @malobre !

**Pull request merged**